src/main/java/de/tum/in/www1/artemis/service/scheduled/EmbeddingTrainingMaterialScheduleService.java

```diff
@@ -0,0 +1,79 @@
+package de.tum.in.www1.artemis.service.scheduled;
+
+import static java.time.Instant.now;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.env.Environment;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+import de.tum.in.www1.artemis.service.connectors.EmbeddingTrainingMaterialService;
+
+@Service
+@Profile("automaticText")
+public class EmbeddingTrainingMaterialScheduleService {
+
+    private final Logger log = LoggerFactory.getLogger(EmbeddingTrainingMaterialScheduleService.class);
+
+    private final Map<Long, ScheduledFuture> scheduledClusteringTasks = new HashMap<>();
+
+    private final EmbeddingTrainingMaterialService embeddingTrainingMaterialService;
+
+    private final Environment env;
```

Comment:
Codacy found an issue: [Perhaps 'env' could be replaced by a local variable.](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968860)